v2/pkg/subscraping/sources/redhuntlabs/redhuntlabs.go

```diff
@@ -0,0 +1,143 @@
+// Package redhuntlabs logic
+package redhuntlabs
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+
+	"github.com/projectdiscovery/subfinder/v2/pkg/subscraping"
+)
+
+type Response struct {
+	Subdomains []string         `json:"subdomains"`
+	Metadata   ResponseMetadata `json:"metadata"`
+}
+
+type ResponseMetadata struct {
+	ResultCount int `json:"result_count"`
+	PageSize    int `json:"page_size"`
+	PageNumber  int `json:"page_number"`
+}
+
+type Source struct {
+	apiKeys   []string
+	timeTaken time.Duration
+	errors    int
+	results   int
+	skipped   bool
+}
+
+func (s *Source) Run(ctx context.Context, domain string, session *subscraping.Session) <-chan subscraping.Result {
+	results := make(chan subscraping.Result)
+	s.errors = 0
+	s.results = 0
+	pageSize := 1000
+	go func() {
+		defer func(startTime time.Time) {
+			s.timeTaken = time.Since(startTime)
+			close(results)
+		}(time.Now())
+
+		randomCred := subscraping.PickRandom(s.apiKeys, s.Name())
+		if randomCred == "" || !strings.Contains(randomCred, ":") {
+			s.skipped = true
+			return
+		}
+
+		creds := strings.Split(randomCred, ":")
+		baseUrl := creds[0] + ":" + creds[1]
+		getUrl := baseUrl + "?domain=" + domain + "&page=1&page_size=" + strconv.Itoa(pageSize)
+		resp, err := session.Get(ctx, getUrl, "", map[string]string{
+			"X-BLOBR-KEY": creds[2], "User-Agent": "subfinder",
+		})
+		if err != nil {
+			results <- subscraping.Result{Source: s.Name(), Type: subscraping.Error, Error: err}
```

Comment:
Do we need both of these errors?  If not, let's remove one of them; I expect API to return the error we're getting in the response.

Reply:
We might require both the errors considering that the API at the moment would return generic ones, and the additional error statement I added was to make it more detailed in case the resources/credits for the subscription plan are exhausted. Open to thoughts/ideas

Reply:
We can combine these errors only if we confirm it's a rate limit error. We don't check or know if it's a rate limit error. What do you think?

Reply:
I tried printing the error it gets while making the request within redhuntlabs.go but for some reason it couldn't hence I had added the second error printing statement. I assume that the error is being created through the function that we have declared for making requests which is outside of the sources directory. Would need some additional help around this one.

Reply:
@umair-rhl,
How about something like this?
```
if err != nil {
			results <- subscraping.Result{Source: s.Name(), Type: subscraping.Error, Error: fmt.Errorf("Encountered error: %v; Note: if you get a 'limit has been reached' error, head over to https://devportal.redhuntlabs.com", err)}
			session.DiscardHTTPResponse(resp)
			s.errors++
			return
		}
```

Reply:
Yeah this seems like a much better solution, [fixed](https://github.com/projectdiscovery/subfinder/pull/978/commits/c8d0e5e109c77954fca0e8f7ab36f279ac9769a8)!